backend/tests/server.spec.js

```diff
@@ -1,79 +1,93 @@
-import {createTestClient} from 'apollo-server-testing'
-import Server from '../src/server.js'
-import {InMemoryDataSource, Post, User} from '../src/db.js'
-import {LOGIN_USER, POSTS, SIGNUP_USER, UPVOTE_POST, USERS, WRITE_POST} from "./server.spec.queries";
-
-const users = [
-    new User({
-        name: "Alice",
-        email: "alice@test.de",
-    }),
-    new User({
-        name: "Bob",
-        email: "bob@test.de",
-    }),
-];
-
-const posts = [
-    new Post({
-        title: "post 1",
-        author: users[0],
-    }),
-    new Post({
-        title: "post 2",
-        author: users[0],
-    }),
-    new Post({
-        title: "post 3",
-        author: users[1],
-    })
-]
-
-let db = new InMemoryDataSource(posts, users)
-let server = new Server({
-    dataSources: () => ({db}),
-    context: ({req, res}) => (req, res)
-})
-let {query, mutate} = createTestClient(server)
+import { createTestClient } from 'apollo-server-testing';
+import { ApolloServer } from 'apollo-server';
+import Server from '../src/server';
+import actualContext from '../src/context';
+import neode from '../src/db/neode';
+import User from '../src/db/entities/User';
+import Post from '../src/db/entities/Post';
+
+import { LOGIN_USER, POSTS, SIGNUP_USER, UPVOTE_POST, USERS, WRITE_POST } from "./server.spec.queries";
+
+let query;
+let mutate;
+let reqMock;
+const context = () => actualContext({ req: reqMock });
+const alice = new User({
+    name: 'Alice',
+    email: 'alice@example.org',
+    password: '1234'
+});
+const bob = new User({
+    name: 'Bob',
+    email: 'bob@example.org',
+    password: '4321'
+});
+
+const bobsPost = new Post({ title: "bobsPost", author: bob })
+
+const cleanDatabase = async() => {
```

Comment:
⭐ ⭐ Your software tests are free of side effects.
⭐ Your database is never left in an invalid state.